### DISCUSSION TITLE
Intel vs M1 silicon for scrcpy

### DISCUSSION THREAD

**Discussion by andrew-clark836** [Homebrew/discussions#1319] [Everyday usage]
Hi all, looking for some help here.

I've been running a trash can MacPro for years, and I love it.  I recently decided to swap out to a MacMini M1, just to see how it would compare.....it didn't!

I had installed scrcpy onto the M1 MacMini and, in the meantime, completely wiped and re-installed BigSur on the MacPro.

I decided to revert back to teh MacPro, and thinking I'd be clever, did a machine to machine restore, which worked really well.  However, now I come to run scrcpy in Terminal, I get the following:

zsh: bad CPU type in executbale scrcpy

So I tried a re-install:

brew reinstall scrcpy

and now get the following:

Error: Cannot install on Intel processor in ARM default prefix (/opt/homebrew)!

So next I did an uninstall

brew uninstall scrcpy

Which worked OK.

But when I then try to install:

brew install scrcpy

I get the following:

Error: Cannot install on Intel processor in ARM default prefix (/opt/homebrew)!
Error: 'git' must be installed and in your PATH!
Error: Cannot install on Intel processor in ARM default prefix (/opt/homebrew)!

I'm guessing I need to delete the opt/homebrew directory and start form scratch, but where is it?  I can;t find it anywhere.

Any pointers fromthis eminenet community gratefully accepted.

**Comment by carlocab** [comment]
Your Mac Pro uses an Intel processor, but you copied over your Homebrew installation from your Mac Mini to your Mac Pro. That doesn't quite work.

You can uninstall Homebrew on your Mac Pro using the uninstall script at https://github.com/Homebrew/install. You want to download the uninstall script and run it with the flag `--prefix /opt/homebrew`.

**Reply by andrew-clark836** [reply]
Thanks.  Tried that.  I'm very new to this so obviously have a syntax error somewhere.  This is what I see.

Last login: Tue Apr 20 09:42:15 on ttys000
andrewclark@Mac-Pro ~ % --prefix /opt/homebrew/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/uninstall.sh)"
zsh: command not found: --prefix
andrewclark@Mac-Pro ~ %

**Reply by carlocab** [reply]
You can't use the uninstall snippet at https://github.com/Homebrew/install, as that does not allow you to pass arguments to the uninstall script. You need to download the uninstall script first and then run it. You want to do

```
curl -OL https://raw.githubusercontent.com/Homebrew/install/HEAD/uninstall.sh
chmod +x uninstall.sh
sudo ./uninstall.sh --prefix /opt/homebrew
```
You can delete the uninstall script when you're done. Running `uninstall.sh` with `sudo` will ask you for your password.
